pennylane_qiskit/qiskit_device.py

```diff
@@ -350,6 +350,14 @@ def expval(self, observable, wires, par):
             prob = np.fromiter(self.probabilities(wires=wires).values(), dtype=np.float64)
             return (eigvals @ prob).real
 
+        if self.analytic:
+            # Raise a warning if backend is a hardware simulator
+            warnings.warn("The analytic calculation of expectations and variances "
+                          "is only supported on statevector backends, not on the {}. "
+                          "The obtained result is based on sampling.".
+                          format(self.backend),
+                          UserWarning)
```

Comment:
Should we move the warning to `__init__`? That way,

* It doesn't have to be repeated for `expval` and `var`,
* The user will know straight away and can change it, rather than waiting for the qnode to be evaluated

Reply:
I first went for it to be in the ``__init__.py``. In that case, however, as by default ``analytic=True`` is used and there will be warnings raised for sampling on HW simulators